src/ray/gcs/gcs_server/gcs_placement_group_scheduler.h

```diff
@@ -149,43 +154,94 @@ class LeaseStatusTracker {
                      std::vector<std::shared_ptr<BundleSpecification>> &unplaced_bundles);
   ~LeaseStatusTracker() = default;
 
-  bool MarkLeaseStarted(const ClientID &node_id,
-                        std::shared_ptr<BundleSpecification> bundle);
-  void MarkLeaseReturned(const ClientID &node_id,
-                         std::shared_ptr<BundleSpecification> bundle,
-                         const Status &status);
-  bool IsAllLeaseRequestReturned() const;
-  bool IsLeasingSucceed() const;
+  /// Indicate the tracker that prepare requests are sent to a specific node.
+  ///
+  /// \param node_id Id of a node where prepare request is sent.
+  /// \param bundle Bundle specification the node is supposed to prepare.
+  /// \return False if the prepare phase was already started. True otherwise.
+  bool MarkPreparePhaseStarted(const ClientID &node_id,
+                               std::shared_ptr<BundleSpecification> bundle);
+
+  /// Indicate the tracker that all prepare requests are returned.
+  /// This will update the state to commiting if all prepare requests succeed.
+  /// \param node_id Id of a node where prepare request is returned.
+  /// \param bundle Bundle specification the node was supposed to schedule.
+  /// \param status Status of the prepare response.
+  /// \param void
+  void MarkPreparePhaseDone(const ClientID &node_id,
+                            std::shared_ptr<BundleSpecification> bundle,
+                            const Status &status);
+
+  /// Used to know if all prepare requests are returend.
+  ///
+  /// \return True if all prepare requests are returned
+  bool IsAllPrepareRequestReturned() const;
+
+  /// Used to know if the prepare phase succeed.
+  ///
+  /// \return True if all prepare requests succeed.
+  bool IsPreparePhaseSucceed() const;
+
+  /// Return a placement group this status tracker is associated with.
+  ///
+  /// \return The placement group of this lease status tracker is tracking.
   const std::shared_ptr<GcsPlacementGroup> &GetPlacementGroup() const;
-  const std::vector<std::shared_ptr<BundleSpecification>> &GetUnplacedBundles() const;
-  const std::shared_ptr<BundleLocations> &GetBundleLocations() const;
+
+  /// Return bundles that should be scheduled.
+  ///
+  /// \return List of bundle specification that are supposed to be scheduled.
+  const std::vector<std::shared_ptr<BundleSpecification>> &GetBundlesToSchedule() const;
+
+  /// This method returns bundle locations that succeed to prepare resources.
+  ///
+  /// \return Location of bundles that succeed to prepare resources on a node.
+  const std::shared_ptr<BundleLocations> &GetPreparedBundleLocations() const;
+
+  /// Return the leasing state.
+  ///
+  /// \return Leasing state.
   const LeasingState GetLeasingState() const;
+
+  /// Mark that this leasing is cancelled.
+  ///
+  /// \return void.
```

Comment:
/// \return void. --Do we need to add this uniformly, or we don't add it? `MarkCommitPhaseStarted` didn't add this.